bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDVertex.java

```diff
@@ -0,0 +1,269 @@
+package org.dataflowanalysis.analysis.dfd.core;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentHashMap.KeySetView;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.dataflowanalysis.analysis.core.AbstractVertex;
+import org.dataflowanalysis.analysis.core.CharacteristicValue;
+import org.dataflowanalysis.analysis.core.DataFlowVariable;
+import org.dataflowanalysis.dfd.datadictionary.AND;
+import org.dataflowanalysis.dfd.datadictionary.AbstractAssignment;
+import org.dataflowanalysis.dfd.datadictionary.Assignment;
+import org.dataflowanalysis.dfd.datadictionary.BinaryOperator;
+import org.dataflowanalysis.dfd.datadictionary.ForwardingAssignment;
+import org.dataflowanalysis.dfd.datadictionary.Label;
+import org.dataflowanalysis.dfd.datadictionary.LabelReference;
+import org.dataflowanalysis.dfd.datadictionary.LabelType;
+import org.dataflowanalysis.dfd.datadictionary.NOT;
+import org.dataflowanalysis.dfd.datadictionary.OR;
+import org.dataflowanalysis.dfd.datadictionary.Pin;
+import org.dataflowanalysis.dfd.datadictionary.TRUE;
+import org.dataflowanalysis.dfd.datadictionary.Term;
+import org.dataflowanalysis.dfd.dataflowdiagram.Flow;
+import org.dataflowanalysis.dfd.dataflowdiagram.Node;
+
+/**
+ * This class represents a vertex references a node in the dfd model. Multiple dfd vertices may reference the same node
+ */
+public class DFDVertex extends AbstractVertex<Node> {
+    private final Map<Pin, DFDVertex> mapPinToPreviousVertex;
+    private final Map<Pin, Flow> mapPinToInputFlow;
+
+    /**
+     * Creates a new vertex with the given referenced node and pin mappings
+     * @param node Node that is referenced by the vertex
+     * @param mapPinToPreviousVertex Map containing relationships between the pins of the vertex and previous vertices
+     * @param mapPinToInputFlow Map containing relationships between the pins of the vertex and the flows connecting the
+     * node to other vertices
+     */
+    public DFDVertex(Node node, Map<Pin, DFDVertex> mapPinToPreviousVertex, Map<Pin, Flow> mapPinToInputFlow) {
+        super(node);
+        this.mapPinToPreviousVertex = mapPinToPreviousVertex;
+        this.mapPinToInputFlow = mapPinToInputFlow;
+    }
+
+    /**
+     * Evaluates the given vertex by determining incoming and outgoing data flow variables and node characteristics
+     */
+    @Override
+    public void evaluateDataFlow() {
```

Comment:
This method seems to do many different things and should either split up or properly documented

Reply:
Split up into smaller documented methods in b9dcc76fdcb61710b418e303f51b62ab0738af5b